File: leetcode/cpp/921.minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int openCount = 0, closedCount = 0;
        for (const char c: s) {
            if (c == '(') {
                openCount++;
            } else if (openCount > 0) {
                openCount--;
            } else {
                closedCount++;
            }
        }
        return openCount + closedCount;
    }
};

File: leetcode/go/921.minimum-add-to-make-parentheses-valid.go
func minAddToMakeValid(s string) int {
    openCount, closedCount := 0, 0
    for _, c := range s {
        if c == '(' {
            openCount++
        } else if openCount > 0 {
            openCount--
        } else {
            closedCount++
        }
    }
    return openCount + closedCount
}

